jina/types/arrays/memmap.py

```diff
@@ -98,11 +108,11 @@ def _load_header_body(self, mode: str = 'a'):
         )
         self._header_entry_size = 24 + 4 * self._key_length
 
-        self._header_map = {
-            r[0]: (idx, r[1], r[2], r[3])
-            for idx, r in enumerate(tmp)
-            if not np.array_equal((r[1], r[2], r[3]), HEADER_NONE_ENTRY)
-        }
+        self._header_map = OrderedDict()
```

Comment:
Is this still needed or can it be reverted to a generic dict, since we use the `buffer_pool` now? All the `move_to_end` logic inside the DAM could be removed as well.

Reply:
reverted to generic dict